mtags/src/main/scala/scala/meta/internal/pc/Keyword.scala

```diff
@@ -125,8 +135,11 @@ object Keyword {
     Keyword("then")
   )
 
-  private def extendsPred(leadingReverseTokens: Iterator[Token]): Boolean = {
-    leadingReverseTokens.filterNot(t => t.is[Token.Whitespace] || t.is[Token.EOF]).take(3).toList match {
+  private def extendsPred(leadingReverseTokens: Array[Token]): Boolean = {
+    leadingReverseTokens.reverseIterator
```

Comment:
Why did you move reverse here actually?

Reply:
It wasn't about moving reverse but changing Iterator to Array. Now we reverse in the same place as before, so here nothing changes